lib/matplotlib/stackplot.py

```diff
@@ -76,6 +83,14 @@ def stackplot(axes, x, *args,
     else:
         colors = (axes._get_lines.get_next_color() for _ in y)
 
+    if hatch:
+        if isinstance(hatch, Iterable):
```

Comment:
be careful because hatches can be multi-char strings, and strings are Iterable.

I think what you may want is 

```python
if isinstance(hatch, (str, None)):
    hatch = itertools.cycle([hatch])
else:
    hatch = itertools.cycle(hatch)
```

I think that should handle even the None case (also `" "` is not a valid hatch pattern and is causing warnings)

Reply:
Thanks, and good point!  You're right about the strings being iterables, and I think your logic is the right choice (though I had to modify slightly because `None` doesn't count as a type so can't be used in `isinstance`).

Reply:
Ahh, `str | None` would work (but is py3.10+ only so we can't use it).

`builtins.NoneType` or `type(None)` would also work, but is used less commonly.

Type hints allow `None` as shorthand, but not `isinstance`, apparently.